app.py

```diff
@@ -17,10 +17,29 @@ def home() -> str:
     return render_template('home.html')
 
 
+@app.route('/health')
+def health() -> Response:
+    return jsonify({'message': 'OK'}), 200
+
+
 @app.route('/animals', methods=['GET'])
 def index() -> Response:
-    animals = Animal.query.all()
-    return jsonify({"animals": [AnimalResponse.model_validate(animal).model_dump(mode='json') for animal in animals]})
+    search_value = request.args.get('search')
+
+    if search_value:
+        animals = Animal.query.filter_by(name=search_value).all()
```

Comment:
You can filter animals by part of animal name (means substring)
Model.query.filter(Model.columnName.contains('sub_string'))

Reply:
Thanks!) Updated query